### PR TITLE
util-linux 2.32 (new formula)

### PR DESCRIPTION
**Pull request by Habbie**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This formula currently installs these binaries:
```
$ brew ls -v util-linux | egrep -w 'bin|sbin' | sort
/usr/local/Cellar/util-linux/2.32/bin/cal
/usr/local/Cellar/util-linux/2.32/bin/col
/usr/local/Cellar/util-linux/2.32/bin/colcrt
/usr/local/Cellar/util-linux/2.32/bin/colrm
/usr/local/Cellar/util-linux/2.32/bin/column
/usr/local/Cellar/util-linux/2.32/bin/getopt
/usr/local/Cellar/util-linux/2.32/bin/hexdump
/usr/local/Cellar/util-linux/2.32/bin/ipcmk
/usr/local/Cellar/util-linux/2.32/bin/isosize
/usr/local/Cellar/util-linux/2.32/bin/logger
/usr/local/Cellar/util-linux/2.32/bin/look
/usr/local/Cellar/util-linux/2.32/bin/mcookie
/usr/local/Cellar/util-linux/2.32/bin/mesg
/usr/local/Cellar/util-linux/2.32/bin/more
/usr/local/Cellar/util-linux/2.32/bin/namei
/usr/local/Cellar/util-linux/2.32/bin/rename
/usr/local/Cellar/util-linux/2.32/bin/renice
/usr/local/Cellar/util-linux/2.32/bin/rev
/usr/local/Cellar/util-linux/2.32/bin/scriptreplay
/usr/local/Cellar/util-linux/2.32/bin/setsid
/usr/local/Cellar/util-linux/2.32/bin/ul
/usr/local/Cellar/util-linux/2.32/bin/whereis
/usr/local/Cellar/util-linux/2.32/sbin/blkid
/usr/local/Cellar/util-linux/2.32/sbin/findfs
/usr/local/Cellar/util-linux/2.32/sbin/fsck.cramfs
/usr/local/Cellar/util-linux/2.32/sbin/fsck.minix
/usr/local/Cellar/util-linux/2.32/sbin/mkfs
/usr/local/Cellar/util-linux/2.32/sbin/mkfs.bfs
/usr/local/Cellar/util-linux/2.32/sbin/mkfs.cramfs
/usr/local/Cellar/util-linux/2.32/sbin/mkfs.minix
/usr/local/Cellar/util-linux/2.32/sbin/mkswap
/usr/local/Cellar/util-linux/2.32/sbin/nologin
/usr/local/Cellar/util-linux/2.32/sbin/swaplabel
/usr/local/Cellar/util-linux/2.32/sbin/wipefs
```

while my mac already has:
```
$ for f in $(brew ls -v util-linux | egrep -w 'bin|sbin' | xargs -n1 basename) ; do ls {/usr,}/{s,}bin/$f ; done  2>/dev/null
/usr/bin/cal
/usr/bin/col
/usr/bin/colcrt
/usr/bin/colrm
/usr/bin/column
/usr/bin/getopt
/usr/bin/hexdump
/usr/bin/logger
/usr/bin/look
/usr/bin/mesg
/usr/bin/more
/usr/bin/renice
/usr/bin/rev
/usr/bin/ul
/usr/bin/whereis
/sbin/nologin
```

What should we do with these? Remove them from the formula? Attempt to prefix them?

It also ships these libraries:
```
/usr/local/Cellar/util-linux/2.32/lib/libblkid.1.dylib
/usr/local/Cellar/util-linux/2.32/lib/libblkid.a
/usr/local/Cellar/util-linux/2.32/lib/libblkid.dylib
/usr/local/Cellar/util-linux/2.32/lib/libsmartcols.1.dylib
/usr/local/Cellar/util-linux/2.32/lib/libsmartcols.a
/usr/local/Cellar/util-linux/2.32/lib/libsmartcols.dylib
/usr/local/Cellar/util-linux/2.32/lib/pkgconfig/blkid.pc
/usr/local/Cellar/util-linux/2.32/lib/pkgconfig/smartcols.pc
```

### CODE REVIEW

**Comment by fxcoudert** [comment]
nologin is present as `/sbin/nologin`

For many of these, I wonder if they actually work (and how they could be used) on a macOS system. For example, the swap and filesystem-related utilities. Is util-linux's `fsck` (or `mkfs`) usable on macOS? Or `mkswap`/`swaplabel` given that macOS has no swap partitions?

**Comment by Habbie** [comment]
> nologin is present as /sbin/nologin

Thanks, missed that.



> For many of these, I wonder if they actually work (and how they could be used) on a macOS system. For example, the swap and filesystem-related utilities. Is util-linux's fsck (or mkfs) usable on macOS? Or mkswap/swaplabel given that macOS has no swap partitions?

Yes, good questions. I'll go through the binaries, see how useful they are.

**Comment by Habbie** [comment]
I have pushed a second commit disabling anything that already ships with macOS. Of the remaining tools I have confirmed that they are not entirely useless on macOS.

Once we agree on this content, I'm happy to squash, of course.

**Comment by Habbie** [comment]
Full file listing at this point:

```
$ brew ls -v util-linux
/usr/local/Cellar/util-linux/2.32/.brew/util-linux.rb
/usr/local/Cellar/util-linux/2.32/AUTHORS
/usr/local/Cellar/util-linux/2.32/bin/ipcmk
/usr/local/Cellar/util-linux/2.32/bin/isosize
/usr/local/Cellar/util-linux/2.32/bin/mcookie
/usr/local/Cellar/util-linux/2.32/bin/namei
/usr/local/Cellar/util-linux/2.32/bin/rename
/usr/local/Cellar/util-linux/2.32/bin/scriptreplay
/usr/local/Cellar/util-linux/2.32/bin/setsid
/usr/local/Cellar/util-linux/2.32/ChangeLog
/usr/local/Cellar/util-linux/2.32/COPYING
/usr/local/Cellar/util-linux/2.32/include/blkid/blkid.h
/usr/local/Cellar/util-linux/2.32/INSTALL_RECEIPT.json
/usr/local/Cellar/util-linux/2.32/lib/libblkid.1.dylib
/usr/local/Cellar/util-linux/2.32/lib/libblkid.a
/usr/local/Cellar/util-linux/2.32/lib/libblkid.dylib
/usr/local/Cellar/util-linux/2.32/lib/pkgconfig/blkid.pc
/usr/local/Cellar/util-linux/2.32/NEWS
/usr/local/Cellar/util-linux/2.32/README
/usr/local/Cellar/util-linux/2.32/sbin/blkid
/usr/local/Cellar/util-linux/2.32/sbin/findfs
/usr/local/Cellar/util-linux/2.32/sbin/fsck.cramfs
/usr/local/Cellar/util-linux/2.32/sbin/fsck.minix
/usr/local/Cellar/util-linux/2.32/sbin/mkfs
/usr/local/Cellar/util-linux/2.32/sbin/mkfs.bfs
/usr/local/Cellar/util-linux/2.32/sbin/mkfs.cramfs
/usr/local/Cellar/util-linux/2.32/sbin/mkfs.minix
/usr/local/Cellar/util-linux/2.32/sbin/mkswap
/usr/local/Cellar/util-linux/2.32/sbin/swaplabel
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/blkid
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/findfs
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/fsck.cramfs
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/fsck.minix
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/ipcmk
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/isosize
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/mcookie
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/mkfs
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/mkfs.bfs
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/mkfs.cramfs
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/mkfs.minix
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/mkswap
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/namei
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/rename
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/scriptreplay
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/setsid
/usr/local/Cellar/util-linux/2.32/share/bash-completion/completions/swaplabel
/usr/local/Cellar/util-linux/2.32/share/doc/util-linux/getopt/getopt-parse.bash
/usr/local/Cellar/util-linux/2.32/share/doc/util-linux/getopt/getopt-parse.tcsh
/usr/local/Cellar/util-linux/2.32/share/man/man1/ipcmk.1
/usr/local/Cellar/util-linux/2.32/share/man/man1/mcookie.1
/usr/local/Cellar/util-linux/2.32/share/man/man1/namei.1
/usr/local/Cellar/util-linux/2.32/share/man/man1/rename.1
/usr/local/Cellar/util-linux/2.32/share/man/man1/scriptreplay.1
/usr/local/Cellar/util-linux/2.32/share/man/man1/setsid.1
/usr/local/Cellar/util-linux/2.32/share/man/man3/libblkid.3
/usr/local/Cellar/util-linux/2.32/share/man/man5/terminal-colors.d.5
/usr/local/Cellar/util-linux/2.32/share/man/man8/blkid.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/findfs.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/fsck.cramfs.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/fsck.minix.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/isosize.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/mkfs.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/mkfs.bfs.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/mkfs.cramfs.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/mkfs.minix.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/mkswap.8
/usr/local/Cellar/util-linux/2.32/share/man/man8/swaplabel.8
```

File: Formula/util-linux.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                          "--prefix=#{prefix}",
+                          "--disable-ipcs",        # does not build on macOS
+                          "--disable-ipcrm",       # does not build on macOS
+                          "--disable-wall",        # already comes with macOS
+                          "--disable-libuuid",     # conflicts with ossp-uuid
+                          "--disable-libsmartcols" # macOS already ships 'column'
+
+    system "make", "install"
+
+    ["cal", # these already come with macOS
```
Comment (fxcoudert) [inline Formula/util-linux.rb]: Can we use the array as `%w[cal col colcrt …`  without line breaks every time?
And move the comment above the line, saying `Remove binaries already shipped by macOS`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/util-linux.rb (hunk 1 above)
Comment (Habbie) [reply Formula/util-linux.rb]: pushed

**Review by fxcoudert — APPROVED** [review]

**Comment by glensc** [comment]
what about bash-completions, shouldn't these be symlinked to bash completions dir? or at least some caveat being displayed how to add those to bash-completion extra dirs?

**Comment by Habbie** [comment]
> what about bash-completions, shouldn't these be symlinked to bash completions dir? or at least some caveat being displayed how to add those to bash-completion extra dirs?

Oh, perhaps! You should PR one of those things :)

**Comment by glensc** [comment]
I don't know how bash completions should be handed. Haven't looked other formulas nor digged docs.

**Pull request closed without merging**